astropy/utils/iers/iers.py

```diff
@@ -802,3 +825,250 @@ def _substitute_iers_b(cls, table):
+
+    @classmethod
+    def open(cls, file=None, cache=False):
+        """Open a leap-second list.
+
+        Parameters
+        ----------
+        file : path, 'erfa', or None
+            Full local or network path to the file holding leap-second data,
+            for passing on to the various ``from_`` class methods.
+            If 'erfa', return the data built-in to the ERFA library.
+            If `None`, use default locations from file and configuration to
+            find a table that is not expired.
+        cache : bool
+            Whether to use cache. Defaults to False, since leap-second files
+            are regularly updated.
+
+        Returns
+        -------
+        leap_seconds : `~astropy.utils.iers.LeapSeconds`
+            Table with 'year', 'month', and 'tai_utc' columns, plus possibly
+            others.
+
+        Notes
+        -----
+        Bulletin C is released about 10 days after a possible leap second is
+        introduced, i.e., mid-January or mid-July.  Expiration days are thus
+        generally at least 150 days after the present.  For the auto-loading,
+        a list comprised of the table shipped with astropy, and files and
+        URLs in ``~astropy.utils.iers.conf`` are tried, returning the first
+        that is sufficiently new, or the newest among them all.
+        """
+        if file is None:
+            return cls.auto_open()
+
+        if file.lower() == 'erfa':
+            return cls.from_erfa()
+
+        if urlparse(file).netloc:
+            file = download_file(file, cache=cache)
+
+        # Just try both reading methods.
+        try:
+            return cls.from_iers_leap_seconds(file)
+        except Exception:
+            return cls.from_leap_seconds_list(file)
+
+    @classmethod
+    def auto_open(cls, files=None):
+        """Attempt to get an up-to-date leap-second list.
+
+        The routine will try the files in sequence until it finds one
+        whose expiration date is "good enough" (see below).  If none
+        are good enough, it returns the one with the most recent expiration
+        date, warning if that file is expired.
+
+        For remote files that are cached already, the cached file is tried
+        first before attempting to retrieve it again.
+
+        Parameters
+        ----------
+        files : list of path, optional
+            List of files/URLs to attempt to open.  By default, uses
+            ``cls._auto_open_files``.
+
+        Returns
+        -------
+        leap_seconds : `~astropy.utils.iers.LeapSeconds`
+            Up to date leap-second table
+
+        Notes
+        -----
+        Bulletin C is released about 10 days after a possible leap second is
+        introduced, i.e., mid-January or mid-July.  Expiration days are thus
+        generally at least 150 days after the present.  We look for a file
+        that expires more than 180 - `~astropy.utils.iers.conf.auto_max_age`
```

Comment:
```suggestion
        that expires more than 180 - `~astropy.utils.iers.Conf.auto_max_age`
```

Reply:
``Conf``  is upper case in the API docs.

Reply:
Thanks, fixed - hopefully got all now (local docs builds are still painful...)